src/main/java/org/broadinstitute/consent/http/service/dao/DacServiceDAO.java

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.consent.http.service.dao;
+
+import com.google.inject.Inject;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Objects;
+import org.broadinstitute.consent.http.db.DaaDAO;
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DataAccessAgreement;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.Update;
+
+public class DacServiceDAO implements ConsentLogger {
+
+  private final Jdbi jdbi;
+  private final DaaDAO daaDAO;
+
+  @Inject
+  public DacServiceDAO(Jdbi jdbi, DaaDAO daaDAO) {
+    this.jdbi = jdbi;
+    this.daaDAO = daaDAO;
+  }
+
+  public void deleteDacAndDaas(Dac dac)
+      throws IllegalArgumentException, SQLException {
+    // fail fast
+    if (Objects.isNull(dac)) {
+      throw new IllegalArgumentException("Invalid DAC");
+    }
+    List<DataAccessAgreement> daas = daaDAO.findAll();
+    List<DataAccessAgreement> filteredDaas = daas.stream()
+        .filter(daa -> daa.getInitialDacId().equals(dac.getDacId())).toList();
+    jdbi.useHandle(handle -> {
+      handle.getConnection().setAutoCommit(false);
+
+      jdbi.useTransaction(handler -> {
+        final String deleteFromLcDaa = "DELETE FROM lc_daa WHERE daa_id in (SELECT daa_id FROM data_access_agreement WHERE initial_dac_id = :dacId)";
+        final String deleteFromDacDaa = "DELETE FROM dac_daa WHERE dac_id = :dacId";
+        final String deleteFromDaa = "DELETE FROM data_access_agreement WHERE initial_dac_id = :dacId";
+        final String deleteMembers = "DELETE FROM user_role WHERE dac_id = :dacId";
+        final String updateDatasets = "UPDATE dataset SET dac_id = null, needs_approval = true WHERE dac_id = :dacId";
+        final String deleteDac = "DELETE FROM dac where dac_id = :dacId";
+
+        Update lcDaaDeletion = handler.createUpdate(deleteFromLcDaa);
+        lcDaaDeletion.bind("dacId", dac.getDacId());
+        lcDaaDeletion.execute();
+
+        Update dacDaaDeletion = handler.createUpdate(deleteFromDacDaa);
+        dacDaaDeletion.bind("dacId", dac.getDacId());
+        dacDaaDeletion.execute();
+
+        filteredDaas.forEach(id -> {
+          Update daaDeletion = handler.createUpdate(deleteFromDaa);
+          daaDeletion.bind("dacId", dac.getDacId());
+          daaDeletion.execute();
+        });
```

Comment:
This can be a single line delete. In this construction, we have N DAAs all with a dac id of, for example, 10. We will be making N delete calls that are exactly the same: `DELETE FROM dac_daa WHERE dac_id = :dacId`